impl_core/src/main/java/io/opencensus/implcore/stats/MeasureToViewMap.java

```diff
@@ -57,35 +61,41 @@
   @GuardedBy("this")
   private final Map<String, Measure> registeredMeasures = Maps.newHashMap();
 
+  // Cached set of exported views. It must be set to null whenever a view is registered or
+  // unregistered.
+  @Nullable private volatile Set<View> exportedViews;
+
   /** Returns a {@link ViewData} corresponding to the given {@link View.Name}. */
   synchronized ViewData getView(View.Name viewName, Clock clock, StatsCollectionState state) {
     MutableViewData view = getMutableViewData(viewName);
     return view == null ? null : view.toViewData(clock.now(), state);
   }
 
-  @Nullable
-  private synchronized MutableViewData getMutableViewData(View.Name viewName) {
-    View view = registeredViews.get(viewName);
-    if (view == null) {
-      return null;
+  Set<View> getExportedViews() {
+    Set<View> views = exportedViews;
+    if (views == null) {
+      synchronized (this) {
+        exportedViews =
+            views = Collections.unmodifiableSet(filterExportedViews(registeredViews.values()));
```

Comment:
Double wrap with unmodifiableSet. Please remove one of them (filterExportedViews also wraps with unmodifiableSet).

Reply:
Oops!  I removed one call.